src/parachain/loans.ts

```diff
@@ -675,13 +677,120 @@ export class DefaultLoansAPI implements LoansAPI {
         return loanAssets;
     }
 
-    async getAccruedRewardsOfAccount(accountId: AccountId): Promise<MonetaryAmount<CurrencyExt>> {
-        const [rewardAccrued, rewardCurrency] = await Promise.all([
+    async _getLatestSupplyIndex(
+        underlyingCurrencyId: CurrencyId,
+        lendTokenId: CurrencyId,
+        currentBlockNumber: number
+    ): Promise<Big> {
+        const [marketSupplyState, marketSupplySpeed, totalIssuance] = await Promise.all([
+            this.api.query.loans.rewardSupplyState(underlyingCurrencyId),
+            // Total KINT / INTR tokens awarded per block to suppliers of this market
+            this.api.query.loans.rewardSupplySpeed(underlyingCurrencyId),
+            this.api.query.tokens.totalIssuance(lendTokenId),
+        ]);
+        const lastIndex = Big(marketSupplyState.index.toString());
+        const supplyRewardSpeed = Big(marketSupplySpeed.toString());
+        const totalSupply = Big(totalIssuance.toString());
+
+        const deltaBlocks = currentBlockNumber - marketSupplyState.block.toNumber();
+        const deltaIndex = supplyRewardSpeed.div(totalSupply).mul(deltaBlocks);
```

Comment:
If the `totalSupply` is very large, the division might be rounded to zero. You should first multiply and then divide (in the other function as well)
```suggestion
        const deltaIndex = supplyRewardSpeed.mul(deltaBlocks).div(totalSupply);
```

Reply:
Great suggestion, thanks!